--- /Users/tamaneko/upside/Project-BearMoon/uniswap-original/v2-core/contracts/UniswapV2Factory.sol	2025-05-15 10:53:14
+++ /Users/tamaneko/upside/Project-BearMoon/kodiak-contracts/UniswapV2Factory/src/pools/v2-core/UniswapV2Factory.sol	2025-05-13 12:16:48
@@ -1,4 +1,4 @@
-pragma solidity =0.5.16;
+pragma solidity =0.8.19;
 
 import './interfaces/IUniswapV2Factory.sol';
 import './UniswapV2Pair.sol';
@@ -10,9 +10,7 @@
     mapping(address => mapping(address => address)) public getPair;
     address[] public allPairs;
 
-    event PairCreated(address indexed token0, address indexed token1, address pair, uint);
-
-    constructor(address _feeToSetter) public {
+    constructor(address _feeToSetter) {
         feeToSetter = _feeToSetter;
     }
 
