--- /Users/tamaneko/upside/Project-BearMoon/uniswap-original/v3-core/contracts/UniswapV3PoolDeployer.sol	2025-05-15 10:53:20
+++ /Users/tamaneko/upside/Project-BearMoon/kodiak-contracts/UniswapV3Factory/src/pools/v3-core/UniswapV3PoolDeployer.sol	2025-05-13 12:15:27
@@ -1,4 +1,4 @@
-// SPDX-License-Identifier: BUSL-1.1
+// SPDX-License-Identifier: GPL-2.0-or-later
 pragma solidity =0.7.6;
 
 import './interfaces/IUniswapV3PoolDeployer.sol';
@@ -35,4 +35,4 @@
         pool = address(new UniswapV3Pool{salt: keccak256(abi.encode(token0, token1, fee))}());
         delete parameters;
     }
-}
+}
\ No newline at end of file
